modules/Core/pages/panel/general_settings.php

```diff
@@ -228,6 +219,32 @@
 $displaynames = Settings::get('displaynames');
 $method = Settings::get('login_method');
 
+$homepage_pages = [[
+    'value' => 'news',
+    'name' => $language->get('admin', 'homepage_news'),
+    'module' => 'Core'
+], [
+    'value' => 'portal',
+    'name' => $language->get('admin', 'portal'),
+    'module' => 'Core'
+], [
+    'value' => 'custom',
+    'name' => $language->get('admin', 'custom_content'),
+    'module' => 'Core'
+]];
+
+foreach ($pages->returnPages() as $key => $page) {
+    if (str_contains($key, '/panel/') || str_contains($key, '/queries/') || str_contains($key, '/user/')) {
+        continue;
+    }
+
+    $homepage_pages[] = [
+        'value' => $key,
+        'name' => $key,
```

Comment:
Should the name/key be purified before passing to the template

Reply:
Solved